lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -595,6 +597,15 @@ data ApiPostRandomAddressData = ApiPostRandomAddressData
     , addressIndex :: !(Maybe (ApiT (Index 'Hardened 'AddressK)))
     } deriving (Eq, Generic, Show)
 
+data ApiWalletMigrateData (n :: NetworkDiscriminant) = ApiWalletMigrateData
+    { passphrase :: !(ApiT (Passphrase "raw"))
+    , addresses :: ![(ApiT Address, Proxy n)]
+    } deriving (Eq, Generic, Show)
```

Comment:
Hi Pawel!

I'd like to suggest the following name:

```suggestion
data ApiWalletMigrationParams (n :: NetworkDiscriminant) = ApiWalletMigrationParams
    { passphrase :: !(ApiT (Passphrase "raw"))
    , addresses :: ![(ApiT Address, Proxy n)]
    } deriving (Eq, Generic, Show)
```

Justification:
1. When reading the terms `MigrateData` and `MigrationInfo`, it's not immediately clear that one of these is a set of parameters, and one of these is a result value. Using the suffix `Params` for the parameters object makes this distinction more obvious. 
2. Related data types should probably share a common prefix (`ApiWalletMigration`).

Reply:
I would be 100% in favor of *Params if it was used throughout the codebase. but we have 
```
, ByronWalletPostData (..)
    , SomeByronWalletPostData (..)
    , ByronWalletFromXPrvPostData (..)
    , ByronWalletPutPassphraseData (..)
    , ApiPostRandomAddressData (..)
    , AccountPostData (..)
    , WalletOrAccountPostData (..)
    , WalletPostData (..)
    , WalletPutData (..)
    , WalletPutPassphraseData (..)
    , PostTransactionData (..)
    , PostTransactionFeeData (..)
    , PostExternalTransactionData (..)
    , ApiSelectCoinsData (..)
```

*Data is kinda convention...

Reply:
> *Data is kinda convention...

Fair enough!

However, in virtually all the other cases, the name includes the HTTP method being used, and this makes it obvious that it is a request. For example:

- `AccountPostData` (`POST`)
- `WalletPutData` (`PUT`)

I think we could also adopt the same convention here, which would give us: 

- `ApiWalletMigrationPostData`
- `ApiWalletMigrationInfo`

This makes it obvious which one is the request data type.

Do you agree?

Reply:
`ApiWalletMigrationPostData` sounds good and even more in line with current convention

Reply:
did in https://github.com/input-output-hk/cardano-wallet/pull/1664/commits/cb4f1a7d89572445ddd6e95bec27780e6c6a6d09

Reply:
Great, thanks!